internal/component/prometheus/operator/common/crdmanager.go

```diff
@@ -364,18 +364,21 @@ func (c *crdManager) configureInformers(ctx context.Context, informers cache.Inf
 	var informer cache.Informer
 	var err error
 
+	timeoutCtx, cancel := context.WithTimeout(ctx, c.args.InformerSyncTimeout)
+	done, _ := timeoutCtx.Deadline()
+	defer cancel()
 	backoff := backoff.New(
-		ctx,
+		timeoutCtx,
 		backoff.Config{
 			MinBackoff: 1 * time.Second,
 			MaxBackoff: 10 * time.Second,
-			MaxRetries: 3, // retry up to 3 times
+			MaxRetries: 0, // Will retry until InformerSyncTimeout is reached
 		},
 	)
 	for backoff.Ongoing() {
 		// Retry to get the informer in case of a timeout.
 		informer, err = getInformer(ctx, informers, prototype, c.args.InformerSyncTimeout)
-		if err == nil {
+		if err == nil || time.Now().After(done) {
```

Comment:
Not sure we need to do this check here. We create a context that will cancel backoff after some time. When that happens `backoff.Ongoing()` will be false.

Then I am pretty sure we need to check `backoff.Errs()`. In this setup this function will only return an error if it was canceled.

Reply:
The intent here was to not log that we were going to retry if we had already exceeded the timeout before the next iteration checks ongoing, but I suppose we can just check Errs instead 👍